core-plugins/src/main/java/io/cdap/plugin/error/ErrorCollector.java

```diff
@@ -40,28 +39,37 @@
 public class ErrorCollector extends ErrorTransform<StructuredRecord, StructuredRecord> {
   private final Config config;
 
+  private static final String MESSAGE_FIELD = "messageField";
+  private static final String CODE_FIELD = "codeField";
+  private static final String STAGE_FIELD = "stageField";
+
   public ErrorCollector(Config config) {
     this.config = config;
   }
 
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (config.messageField != null && inputSchema.getField(config.messageField) != null) {
-        throw new InvalidConfigPropertyException(String.format(
-          "Input schema already contains message field '%s'. Please set message field to a different value.",
-          config.messageField), "messageField");
+        collector.addFailure(
+            String.format("Input schema already contains message field '%s'.", config.messageField),
+            "Please set message field to a different value.")
```

Comment:
as we're going through these messages, we've stopped putting 'Please', as it's weird to have it in some places but not others, and sometimes it doesn't make sense.

Set the message field to a different value

Reply:
similar comments below

Reply:
Fixed